src/producers.rs

```diff
@@ -132,9 +133,13 @@ async fn produce(
 
         total_size += sz;
         let fut = producer.send(
-            FutureRecord::to(&topic).key(&key).payload(payload_slice),
+            FutureRecord::to(&topics[current_producer_index])
+                .key(&key)
+                .payload(payload_slice),
             timeout,
         );
+        current_producer_index = (current_producer_index + 1) % topics.len();
```

Comment:
Can't this pattern lead to any producer producing to all topics over time?

Reply:
I don't believe so. `topics` in this case only contains the subset of topics a given producer was assigned.